### PR TITLE
[OnePieceTube] Add new extractor

### PR DESCRIPTION
**Pull request by opinator**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

An extractor for the site onepiece-tube.com

### CODE REVIEW

File: youtube_dl/extractor/onepiecetube.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import lxml.html
```
Comment (dstftw) [inline youtube_dl/extractor/onepiecetube.py]: `lxml` is not allowed.

File: youtube_dl/extractor/onepiecetube.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            'title': 'One Piece 778 - Zur Weltkonferenz! Rebecca und das Sakura Königreich!',
+            'series': 'One Piece',
+            'episode': '778',
+            'thumbnail': 'http://cdn.ani-stream.com/thumb/4rt9j41nplak.jpg'
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, 'OnePiece-Tube')
```
Comment (dstftw) [inline youtube_dl/extractor/onepiecetube.py]: Second parameter must be video id.

File: youtube_dl/extractor/onepiecetube.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        title = None
+
+        # Try HTML title
+        for elem in root.xpath('//head/title/text()'):
+            title = elem
+            break
+
+        # Try meta information title
+        if title is None:
+            for elem in root.xpath('//meta[@name="title"]/@content'):
```
Comment (dstftw) [inline youtube_dl/extractor/onepiecetube.py]: `utils.xpath_*`.

File: youtube_dl/extractor/onepiecetube.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            break
+
+        # Download embedded webpage
+        webpage = self._download_webpage(embed_url, 'anistream.com')
+
+        # Get real location of file
+        url = self._search_regex(r'file: ["\'](?P<url>https?://[a-zA-Z0-9/\-\.]*?.mp4)["\']', webpage, 'url', group='url')
+
+        # Get Thumbnail
+        thumbnail = 'http:' + self._search_regex(r'image: ["\'](?P<thumb>//[a-zA-Z0-9/\-\.]*?.jpg)["\']', webpage, 'thumb', group='thumb', default=None)
```
Comment (dstftw) [inline youtube_dl/extractor/onepiecetube.py]: `'http:' + None = TypeError`

File: youtube_dl/extractor/onepiecetube.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        embed_url = None
+        for elem in root.xpath('//*[@id="embed"]/iframe/@src'):
+            embed_url = elem
+            break
+
+        # Download embedded webpage
+        webpage = self._download_webpage(embed_url, 'anistream.com')
+
+        # Get real location of file
+        url = self._search_regex(r'file: ["\'](?P<url>https?://[a-zA-Z0-9/\-\.]*?.mp4)["\']', webpage, 'url', group='url')
```
Comment (dstftw) [inline youtube_dl/extractor/onepiecetube.py]: No sense to use named group when it's the only one.

File: youtube_dl/extractor/onepiecetube.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            'episode': '778',
+            'thumbnail': 'http://cdn.ani-stream.com/thumb/4rt9j41nplak.jpg'
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, 'OnePiece-Tube')
+
+        # Compose title information so far
```
Comment (dstftw) [inline youtube_dl/extractor/onepiecetube.py]: Remove all pointless comments.

**Comment by yan12125** [comment]
This website seems distributing copyrighted products: https://www.google.com/transparencyreport/removals/copyright/searchdata/domains/?id=onepiece-tube.com

By the way, all pages on http://onepiece-tube.com/ are 404 from here

**Pull request closed without merging**